src/components/ArcCharts.jsx

```diff
@@ -0,0 +1,178 @@
+        return (
+            <ChartContainer
+                height={height}
+                width={width}
+                config={config}
+                xScale={'linear'}
+                theme={theme}
+                disableAxes
+                disableContainer
+            >
+                <VictoryPie
+                    randomVal={random}
+                    data={pieChartData}
+                    labelComponent={
+                        config.percentage ?
+                            <VictoryLabel text={''} /> :
+                            <VictoryTooltip
+                                orientation='top'
+                                pointerLength={4}
+                                cornerRadius={2}
+                                flyoutStyle={{ fill: '#000', fillOpacity: '0.8', strokeWidth: 0 }}
+                                style={{ fill: '#b0b0b0' }}
+                            />
+                    }
+                    innerRadius={
+                        this.chartMode === 'donut' || config.percentage ?
+                            ((height > width ? width : height) / 4) : 0
+                    }
+                    labels={
+                        config.percentage === true ?
+                            '' :
+                            d => `${d.x} : ${((d.y / (_.sumBy(pieChartData, o => o.y))) * 100).toFixed(2)}%`
+                    }
+                    style={{ labels: { fontSize: 6 } }}
+                    labelRadius={height / 3}
+                    events={[{
+                        target: 'data',
+                        eventHandlers: {
+                            onClick: () => {
+                                return [
+                                    {
+                                        target: 'data',
+                                        mutation: this.handleMouseEvent,
+                                    },
+                                ];
+                            },
+                        },
+                    }]}
+                    animate={
+                        config.animate ?
+                            { onEnter: { duration: 100 } } : null
+                    }
+                />
+                {
+                    config.percentage ?
+                        <VictoryLabel
+                            textAnchor="middle"
+                            x="55%"
+                            y="50%"
+                            text={`${Math.round(pieChartData.length > 0 ? pieChartData[0].y : 0)}%`}
+                            style={{ fontSize: config.labelFontSize || 45, fill: config.labelColor || 'black' }}
+                        /> :
+                        <LegendComponent
+                            height={height}
+                            width={width}
+                            legendItems={pieChartData.map(data => ({ name: data.x, symbol: data.symbol }))}
+                            interaction={() => {}}
```

Comment:
Do we really need this empty function?

Reply:
it would throw an error otherwise on interaction